pkg/cmd/repo/list/http.go

```diff
@@ -0,0 +1,176 @@
+package list
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/api"
+	"github.com/shurcooL/githubv4"
+)
+
+type RepositoryList struct {
+	Repositories []Repository
+	TotalCount   int
+}
+
+func listRepos(client *api.Client, hostname string, limit int, owner string, filter FilterOptions) (*RepositoryList, error) {
+	type reposBlock struct {
+		TotalCount      int
+		RepositoryCount int
+		Nodes           []Repository
+		PageInfo        struct {
+			HasNextPage bool
+			EndCursor   string
+		}
+	}
+
+	type response struct {
+		RepositoryOwner struct {
+			Repositories reposBlock
+		}
+		Search reposBlock
+	}
+
+	fragment := `
+	fragment repo on Repository {
+		nameWithOwner
+		description
+		isFork
+		isPrivate
+		isArchived
+		updatedAt
+	}`
+
+	// If `--archived` wasn't specified, use `repositoryOwner.repositores`
+	query := fragment + `
+	query RepoList($owner: String!, $per_page: Int!, $endCursor: String, $fork: Boolean, $privacy: RepositoryPrivacy) {
+		repositoryOwner(login: $owner) {
+			repositories(
+				first: $per_page,
+				after: $endCursor,
+				privacy: $privacy,
+				isFork: $fork,
+				ownerAffiliations: OWNER,
+				orderBy: { field: UPDATED_AT, direction: DESC }) {
+					totalCount
+					nodes {
+						...repo
+					}
+					pageInfo {
+						hasNextPage
+						endCursor
+					}
+			}
+		}
+	}`
+
+	perPage := limit
+	if perPage > 100 {
+		perPage = 100
+	}
+
+	variables := map[string]interface{}{
+		"per_page":  githubv4.Int(perPage),
+		"endCursor": (*githubv4.String)(nil),
+	}
+
+	hasArchivedFilter := filter.Archived
+
+	if hasArchivedFilter {
+		// If `--archived` was specified, use the `search` API rather than
+		// `repositoryOwner.repositories`
+		query = fragment + `
```

Comment:
[`repositoryOwner.repositories`](https://docs.github.com/en/graphql/reference/interfaces#repositoryowner) doesn't allow to filter repositories based on whether they are archived or not, the only way I found to do that was through the Search API.

Caveat:
It returns an ordered list but the `updatedAt` field isn't consistent with the one provided by `repositoryOwner` and makes the list look like if it were unordered.

`gh`

![Screenshot_2021-02-11_22-14-11](https://user-images.githubusercontent.com/6853656/107719384-0783af00-6cd0-11eb-90ed-23269d4def88.png)

`GitHub web`

![Screenshot_2021-02-11_22-20-38](https://user-images.githubusercontent.com/6853656/107719640-a01a2f00-6cd0-11eb-8fc5-58145efebd03.png)